Added support for Velbus

thomasdelaet opened the pull request:
## Description:

**Pull request in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io) with documentation (if applicable):** home-assistant/home-assistant.github.io#2830

## Example entry for `configuration.yaml` (if applicable):
```yaml
[velbus]
serial_port = "/dev/ttyUSB00"
lights:
  - module: 0xda
    channel: 1
    name: "Kitchen Light"
binary_sensors:
  - module: 0xea
    channel: 2
    name: "Kitchen Switch"

```

## Checklist:

If user exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io)

If the code communicates with devices, web services, or third-party tools:
  - [x] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
  - [x] New dependencies have been added to the `REQUIREMENTS` variable ([example][ex-requir]).
  - [x] New dependencies are only imported inside functions that use them ([example][ex-import]).
  - [x] New dependencies have been added to `requirements_all.txt` by running `script/gen_requirements_all.py`.
  - [x] New files were added to `.coveragerc`.

[ex-requir]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L14
[ex-import]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L54

thomasdelaet committed:
Added support for Velbus

thomasdelaet committed:
Added support for Velbus

homeassistant commented:
Hi @thomasdelaet, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant#7684).

Once you do that we will be able to review and accept this pull request.

Thanks!

thomasdelaet committed:
change import statement of platform

thomasdelaet committed:
change import statement of platform

thomasdelaet committed:
ignore from testing

thomasdelaet committed:
ignore from testing

thomasdelaet committed:
update requirements

thomasdelaet committed:
update requirements

balloob committed:
Update state automation to work with new and deleted state changes

balloob committed:
Version bump to 0.45.1

tobygray committed:
zoneminder: fix incorrect use of logging.exception. (#7675)

Prior to this change the zoneminder component was attempting to
use logging.exception outside of exception handling code. This
would lead to the traceback module throwing an exception when
trying to work out the traceback for the exception.

This fixes the issue by changing the exception call into a
plain error logging call.

andrey-git committed:
Zwave: Apply refresh_node workaround on 1st instance only (#7579)

* Apply refresh_node workaround on 1st instance only

* Add another test

balloob committed:
Guard against new and removed state change events (#7687)

thomasdelaet committed:
fix pylint errors

thomasdelaet committed:
fix pylint errors

turbokongen committed:
ZWave: Add reset service to meters (#7676)

* Add reset service for command_class meters.

* Add reset service for command_class meters.

* cast index to const.py

emlove committed:
Remove defunct INSTALL_OPENZWAVE from Dockerfile (#7697)

azogue committed:
Fix telegram chats (#7689)

* bugfix for Telegram chat_ids

- Negative `chat_id`s for groups.
- Include `chat_id` in event data.
- Handle KeyError when receiving other types of messages, as `new_chat_member` ones, and send them as text.

* unused import

* fix double quote style, fix boolean expr, change warning msg

* mistake

* some more fixes

- fix if condition for msg bad fields.
- return True for a correct but not allowed or not recognized message: if not, the message arrives continuously.
- Allow to receive messages from unauthorized users if they come from authorized groups.

* support for `edited_message`s

- They come as normal messages, except for the 'edited_message' field instead of 'message'.

azogue committed:
Fix telegram chats (#7689)

* bugfix for Telegram chat_ids

- Negative `chat_id`s for groups.
- Include `chat_id` in event data.
- Handle KeyError when receiving other types of messages, as `new_chat_member` ones, and send them as text.

* unused import

* fix double quote style, fix boolean expr, change warning msg

* mistake

* some more fixes

- fix if condition for msg bad fields.
- return True for a correct but not allowed or not recognized message: if not, the message arrives continuously.
- Allow to receive messages from unauthorized users if they come from authorized groups.

* support for `edited_message`s

- They come as normal messages, except for the 'edited_message' field instead of 'message'.

cgtobi committed:
Fix playback control of web streams (#7683)

Web streams can't be paused and resumed later. That's why volumio stops them instead of pausing them.

cgtobi committed:
Fix playback control of web streams (#7683)

Web streams can't be paused and resumed later. That's why volumio stops them instead of pausing them.

amelchio committed:
Sort entities in default groups by name (#7681)

* Sort entities in default groups by name

* Cleanups from review

LvivEchoes committed:
Continue tracking device over dhcp lease table if wireless adapter not installed (#7690)

tobygray committed:
device_tracker.ubus: Handle empty results (#7673)

If OpenWRT isn't running the DHCP server then some OpenWRT hardware,
such as TP-Link TL-WDR3600 v1, can't determine the host
corresponding to an associated wifi client. This change handles that
by returning None when the request has no data in the result.

tobygray committed:
device_tracker.ubus: Handle empty results (#7673)

If OpenWRT isn't running the DHCP server then some OpenWRT hardware,
such as TP-Link TL-WDR3600 v1, can't determine the host
corresponding to an associated wifi client. This change handles that
by returning None when the request has no data in the result.

mezz64 committed:
Bump pyEight version (#7701)

fabaff committed:
Upgrade python-telegram-bot to 6.0.1 (#7704)

fabaff committed:
Make 'sender' as requirement for the config (fixes #7698) (#7706)

balloob committed:
Allow fetching hass.io panel without auth (#7714)

balloob committed:
Allow fetching hass.io panel without auth (#7714)

balloob committed:
Merge pull request #7686 from home-assistant/release-0-45-1

0.45.1

infamy committed:
update apcacccess used in apcupsd to 0.0.10, which fixes random file drop from apcaccess (#7722)

balloob committed:
Constrain requests to a version (#7725)

Add an optional extended description…

aronsky committed:
Asuswrt continuous ssh (#7728)

* Make ssh and telnet connections continuous in asuswrt

* Refactored SSH and Telnet connections into respective classes.

* Fixed several copy-paste typos and errors.

* More typos fixed.

* Small changes to arguments, to pass automated tests.

* Removed unsupported named arguments.

* Fixed a couple of mistakes in Telnet, and other lint errors.

* Added Telnet tests, and added lint exceptions.

* Removed comments from tests, as they irritated the hound.

bzillins committed:
Ensure https base_url in telegram bot (#7726)

antsar committed:
Prevent the random template filter from caching its output. Fixes #5678 (#7716)

nordeep committed:
Don't initialize mqtt components which have already been discovered (#7625)

* Don't initialize mqtt components which have already been discovered

* Fix string length

* Fix blank lines, fix constant name

* Remove globals. Remove JSON dump

* Add tests. Update grammar

* PEP8 style issue

* Add hyphen to object_id regex

* PEP8 style fix

amelchio committed:
LIFX: disable color features for white-only bulbs (#7742)

The product type is already established in order to decide the Kelvin range
so just reuse that information to disable color features for white-only lights.

Also change the breathe/pulse effects to be more useful for white-only
bulbs. For consistency, color bulbs set to a desaturated (i.e. white-ish)
color get the same default treatment as white-only bulbs.

antsar committed:
Don't block startup more than 60 seconds while waiting for components. (#7739)

Commit:
Always return rgb color of bulbs (#7743)

bjw-s committed:
Redesign monitored variables for hp_ilo sensor (#7534)

* Redesign monitored variables

Allow generating specific sensors without the need for template sensors

* Import 3rd party library inside update method

* Remove jsonpath_rw dependency

* Do not interfere with value_template or ilo_data output

Do not interfere with value_template or ilo_data output, this is now the responsibility of the user and should be handled in `configuration.yaml`

Fix UnusedImportStatement

Fix newline after function docstring

* Always output results to state

stu-gott committed:
Fix time_date sensor to update at predictable intervals (#7644)

* Fix time_date sensor to update at predictable intervals

* Delete automations.yaml

everix1992 committed:
Added new commands and functionality to the harmony remote component. (#7113)

* Added new commands and functionality to the harmony remote component.

-This includes the ability to optionally specify a number of times to repeat a specific command, such as pressing the volume button multiple times.
-Also added a new command that allows you to send multiple commands to the harmony at once, such as sending a set of channel numbers.
-Updated the unit tests for these changes.

* Fix flake8 coding violations

* Remove send_commands command and make send_command handle a single or list of commands

* Remove send_commands tests

* Update itach and kira remotes for new send_command structure. Fix pyharmony version in requirements_all.txt

* Fix incorrect variable name

* Fix a couple minor issues with remote tests

cribbstechnologies committed:
Template light (#7657)

* starting light template component

* linting/flaking

* starting unit tests from copypasta

* working on unit testing

* forgot to commit the test

* wrapped up unit testing

* adding remote back

* updates post running tox

* Revert "adding remote back"

This reverts commit 852c87ff9694dfc48e92b74fd9dbafbc164a2393.

* adding submodule back from origin

* updating submodule

* removing a line to commit

* re-adding line

* trying to update line endings

* trying to fix line endings

* trying a different approach

* making requested changes, need to fix tests

* flaking

* union rather than intersect; makes a big difference

* more tests passing, not sure why this one's failing

* got it working

* most of the requested changes

* hopefully done now

* sets; the more you know

balloob committed:
Update links.html

amigian74 committed:
Exclude filter for event types (#7627)

* add exclude filter for event types to recorder component

* corrected long line (279)

* change source code structure
add test for exclude event types

* code cleanup

* change source code structure

* Update __init__.py

* Update test_init.py

hoh committed:
Change setup script to use pip install instead of setup.py develop (#7756)

Using `python setup.py develop` did not manage to install the required dependencies.
This updates `script/setup` to use `pip install -e .` instead in order to resolve the required dependencies.

balloob committed:
Merge branch 'master' into dev

antsar committed:
Add icons to device tracker. (#7759)

emlove committed:
Rename of the zwave hass.data constants (#7768)

* Rename of the zwave hass.data constants

* Remove zwave since it is already implied

balloob committed:
Get rid of mock http component app (#7775)

* Remove mock_http_component from config tests

* Remove mock_http_component_app from emulated hue test

balloob committed:
Remove usage of event_loop fixture (#7776)

balloob committed:
assertEquals is deprecated (#7777)

emlove committed:
Support for GE Zwave fan controller (#7767)

* Support for GE Zwave fan controller

* Tests for zwave fan

* Add additional fan workarounds

balloob committed:
Replace executor with async_add_job (#7658)

* Remove executor

* Lint

* Lint

* Fix tests

azogue committed:
Fix Telegram Bot send file to multiple targets, snapshots of HA cameras, variable templating, digest auth (#7771)

* fix double template rendering when messages come from notify.telegram

* fix 'chat' information not present in callback queries

* better inline keyboards with yaml

To make a row of InlineKeyboardButtons you pass:
- a list of tuples like: `[(text_b1, data_callback_b1), (text_b2, data_callback_b2), ...]
- a string like: `/cmd1, /cmd2, /cmd3`
- or a string like: `text_b1:/cmd1, text_b2:/cmd2`

Example:
```yaml
data:
   message: 'TV is off'
   disable_notification: true
   inline_keyboard:
     - TV ON:/service_call switch.turn_on switch.tv, Other:/othercmd
     - /help, /init
```

* fix send file to multiple targets

* fix message templating, multiple file targets, HA cameras

- Allow templating for caption, url, file, longitude and latitude fields
- Fix send a file to multiple targets
- Load data with some retrying for HA cameras, which return 500 one or two times sometimes (generic cams, always!).
- Doc in services for new inline keyboards yaml syntax: `Text button:/command`

* HttpDigest authentication as proposed in #7396

* review changes

- Don't use `file` as variable name.
- For loop
- Simplify filter allowed `chat_id`s.

* Don't use `file` as variable name!

* make params outside the while loop

* fix chat_id validation when editing sent messages

balloob committed:
Fix more deprecation warnings (#7778)

* Remove setting up an hbmqtt broker

* Don't pass loop to web.Application in tests

* Use .query instead of deprecated .GET for aiohttp requests

* Fix closing file resource

* Do not use asyncio mark

* Notify.html5 - PyJWT: Use options to disable verify

* Yamaha: Test was still using deprecated ip

* Remove pytest-asyncio

sander76 committed:
Powerview to async (#7682)

* first commit

* first commit

* first commit

* first commit

* changing requirements

* updated requirements_all.txt

* various changes as suggested in the comments.

* using global values for dict keys.

CTLS committed:
Fix home/stay in concord232 (#7789)

Commit:
Add Marantz SSDP discovery / Detect error string in AppCommand.xml body (#7779)

fabaff committed:
Upgrade Sphinx to 1.6.2 (#7805)

fabaff committed:
Upgrade slacker to 0.9.50 (#7797)

Cinntax committed:
Bump pyenvisalink to version 2.1 (#7803)

fabaff committed:
Upgrade pysnmp to 4.3.6 (#7806)

Klikini committed:
Rachio (Sprinklers) (#7600)

* Rachio platform started

* Rachio tests

* detect bad api token

* Documentation, Code cleanup

* Docstrings end with a period, log uses %

* Fix arguments, default run time is now 10 minutes

* Fix typo, remove todo (GH issue exists)

* Revert polymer submodule commit

* Use a RachioPy version with SSL cert validation

* Update requirements

Teagan42 committed:
Rest notify data (#7757)

* Rest notify data

* Cleanup

* Fix spaces

aronsky committed:
Added handling of an AssertionError from pxssh failed login (#7750)

* Added handling of an AssertionError from pxssh failed login

* Destory and re-create pxssh instance, to fix behavior upon router restart.

fabaff committed:
Upgrade sqlalchemy to 1.1.10 (#7807)

fabaff committed:
Upgrade mypy to 0.511 (#7809)

Add an optional extended description…

aequitas committed:
Allow configuring DSMR5 protocol. (#7535)

* Allow configuring DSMR5 protocol.

* Give good example.

* Using dev branch until released upstream.

* Update to dsmr_parser supporting v5 arguments.

azogue committed:
Telegram send image: fix mimetype detection (#7802)

* Add `name` var to BytesIO content to get recognized

Sometimes the python-telegram-bot doesn't recognize the mimetype of the file and looks after a name variable to deduce it. Fixes #7413

* bytesio stream recycle less explicit

mezz64 committed:
Bump aiohttp to 2.1.0 (#7825)

pvizeli committed:
Cleanup telegram / Add url to webhook (#7824)

* Cleanup telegram / Add url to webhook

* fix lint

* Fix lint

fabaff committed:
Upgrade pysnmp to 4.3.7 (#7828)

Danielhiversen committed:
Flux led fix (#7829)

* Update flux_led.py

* style fix

happyleavesaoc committed:
update snapcast media player (#7079)

* update snapcast

* fix docstrings

* bump dep version

* address snapcast review comments

* add snapcast group volume support

* fix snapcast requirements

* update snapcast client entity id

* snapshot/restore functions

* refactor snapshot/restore services

* clean up

* update snapcast req

* bump version

* fix async updates

fabaff committed:
Update docstrings and log messages (#7709)

fabaff committed:
Upgrade PyMVGLive to 1.1.4 (#7832)

tchellomello committed:
Added persistent error message if cover.myq fails to load (#7700)

* Show persistent error if cover.myq fails

* Fixed typo on getLogger()

* Added ValueError on except condition

* Make pylint happy

* Removed DEFAULT_ENTITY_NAMESPACE since it is not being used

balloob committed:
Add initial benchmark framework (#7827)

* Add initial benchmark framework

* Use timer from timeit

philhawthorne committed:
Downgrade Docker to Python 3.5 to solve Segmentation Faults (#7799)

Downgrades the Dockerfiles used by Home Assistant to Python 3.5, after
Python 3.6 base image was causing segmentation faults.

See home-assistant/home-assistant#7752

tchellomello committed:
Introduced support to Netgear Arlo Cameras (#7826)

*  Introduced support to Netgear Arlo Cameras

* Using async_setup_platform() and applied other changes

* Removed unecessary variables

* Using asyncio for sensor/arlo

* Update arlo.py

* Removed entity_namespace

fabaff committed:
Allow 'base_url' (fixes #7784) (#7796)

emlove committed:
Expose the node name on the zwave node entity (#7787)

danielperna84 committed:
Bumped pyhomematic version (#7861)

emlove committed:
Rename time trigger 'after' to 'at' (#7846)

emlove committed:
Use expected behvaior for above/below (#7857)

azogue committed:
log time delay of domain setup in info level (#7808)

* log time delay of domain setup in info level

 * when setup problems appear, it's difficult to debug which are the components that took a lot to set up. This minimal change goes further than the 'slow setup warning' and measures the setup time interval for each domain.

* use timer as in helpers/entity

bokub committed:
Optimize history_stats efficiency and database usage (#7858)

abmantis committed:
dont use default for switch name, so that the object id is used (#7845)

amelchio committed:
Disallow ambiguous color descriptors in the light.turn_on schema (#7765)

* Disallow ambiguous color descriptors in the light.turn_on schema

* Update tests

balloob committed:
Make monkey patch work in Python 3.6 (#7848)

* Make monkey patch work in Python 3.6

* Update dockerfiles back to 3.6

* Lint

* Do not set env variable for dockerfile

* Lint

craigjmidwinter committed:
fix permissions issue for Insteon Local #6558 (#7860)

* fix unlinked commit

* Update insteon_local.py

infamy committed:
Start of migration framework, to allow moving of files in the config … (#7740)

* Start of migration framework, to allow moving of files in the config directory to be hidden, ios.conf used as the first one to undergo this change.

* Update const.py

* Update test_config.py

* improvement to syntax

Teagan42 committed:
Spotify aliases (#7702)

* Alias support for spotify devices

* Fix log

* Formatting/Fixes

* Remove default arg

* Add default keyword

* None check

titilambert committed:
Query in InfluxDB sensor is now templatable (#7634)

molobrakos committed:
fix for https://github.com/home-assistant/home-assistant/issues/7019 (#7618)

bjw-s committed:
Check if media commands are actually applicable (#7595)

* Check if media commands are actually applicable

- Explicitly allow ‘stop’ and ‘play’ on radio streams
- Disallow media commands when the playlist is empty
- Check if command is supported when calling `turn_on` and `turn_off`

* Suppress UPnP error 701 on media commands

* Clean up soco_filter_upnperror

Clean up soco_filter_upnperror and fix small bug in support_previous_track determination

DavidLP committed:
Quickfix Bug #7384 (#7582)

* Quickfix Bug #7384

* Fix devices not available runtime bug

aequitas committed:
Align switch group handling with light. (#7577)

balloob committed:
Update netdisco (#7865)

balloob committed:
Update frontend

balloob committed:
Version bump to 0.46

balloob committed:
Version bump to 0.47.0dev0

molobrakos committed:
Update squeezebox.py (#7617)

Do not fail in case no players are connected, in which case squeezeserver will return a result without player_loop.

balloob committed:
Fix html5 unsub (#7874)

* Fix #7758 subscription expiration/removal

Removes a subscription after receiving an HTTP 410 response when trying to send a new message.

* Fix tests failing due to additional call

* Fix code style

* Lint

Mister-Espria committed:
Added effects to Yeelight bulbs (#7152)

* Added effects to Yeelight bulbs

* Fix Typo and Use randint instead of randrange

* Added Effects

* updated requirements_all.txt

* fix empty line

* minor fixes

* fix passing effects as parameter

twendt committed:
Enocean Binary Sensor: Handle click of both rockers (#7770)

emlove committed:
Support for renaming ZWave values (#7780)

* Support for renaming ZWave values

* Improve test

emlove committed:
Support for renaming ZWave values (#7780)

* Support for renaming ZWave values

* Improve test

balloob committed:
Fix telegram_bot (#7877)

balloob committed:
Fix telegram_bot (#7877)

amelchio committed:
[light.lifx] Update aiolifx (#7882)

This makes LIFX Gen3 lights work with the current firmware.

amelchio committed:
[light.lifx] Update aiolifx (#7882)

This makes LIFX Gen3 lights work with the current firmware.

joopert committed:
Change nad_receiver to pypi (#7852)

* Change to pypi

* add requirements

andrey-git committed:
Switch pymyq to pypi (#7884)

fabaff committed:
Use constants (#7888)

fabaff committed:
Upgrade pyasn1-modules to 0.0.9 (#7887)

fabaff committed:
Upgrade sendgrid to 4.2.0 (#7886)

fabaff committed:
Upgrade python-telegram-bot to 6.0.3 (#7885)

fabaff commented on homeassistant/components/binary_sensor/velbus.py:
```diff
@@ -0,0 +1,82 @@
+"""
+Support for Velbus switches.
```
This is the `binary_sensor` platform. We use the term `switch` in a different context.

fabaff commented on homeassistant/components/velbus.py:
```diff
@@ -0,0 +1,71 @@
+"""
+import asyncio
+import logging
+import velbus
```
Please make the import inside the functions which uses the module.

fabaff commented on homeassistant/components/velbus.py:
```diff
@@ -0,0 +1,71 @@
+PLATFORM_SCHEMA = vol.Schema({
+    vol.Required('serial_port'): cv.string,
+    vol.Optional('lights'): vol.All(cv.ensure_list, [CONFIG_MODULE]),
+    vol.Optional('switches'): vol.All(cv.ensure_list, [CONFIG_MODULE])
```
Please use the constants which are available in `const.py`.

fabaff commented on homeassistant/components/velbus.py:
```diff
@@ -0,0 +1,71 @@
+    device = conf.get('serial_port')
+
+    connection = velbus.VelbusUSBConnection(device)
+    controller = velbus.Controller(connection)
```
For the users it would be helpful if the exceptions are caught in case of connection or setup failures. The setup should return with a log entry otherwise the users will end up with a non-functional component.

fabaff commented on homeassistant/components/velbus.py:
```diff
@@ -0,0 +1,71 @@
+"""
+Support for Velbus platform.
+
+For more details about this platform, please refer to the documentation at XXX
```
The link needs to be https://home-assistant.io/components/velbus/

fabaff commented on homeassistant/components/velbus.py:
```diff
@@ -0,0 +1,71 @@
+    hass.data['VelbusController'] = controller
+
+    @callback
+    def stop_velbus(event):  # noqa: D401
```
D401 is an easy fix.

fabaff commented on homeassistant/components/velbus.py:
```diff
@@ -0,0 +1,71 @@
+        async_load_platform(hass, 'light', DOMAIN, conf['lights'], config)
+    )
+    hass.async_add_job(
+        async_load_platform(hass, 'binary_sensor', DOMAIN, conf['switches'],
```
Using the term `switches` while working with `binary_sensor` will lead to confusion.

thomasdelaet replied:
Thanks for all your comments! Very helpful. I'll make those changes + also make sure I submit pull request for documentation

balloob committed:
Fix MQTT camera test (#7878)

balloob committed:
Fix MQTT camera test (#7878)

balloob committed:
Merge pull request #7866 from home-assistant/release-0-46

0.46

Commit:
Added a Taps Aff binary sensor (#7880)

* Added a Taps Aff binary sensor

* PR Review updates

* Added a Taps Aff binary sensor

* PR Review updates

* Improved error handling

* Cosmetic changes (ordering, docstings, etc.)

trisk committed:
lock.sesame: New lock platform for Sesame smart locks (#7873)

* Manage Sesame devices through CANDY HOUSE's cloud API
* Add dependency on new pysesame library

aequitas committed:
Dsmr5 revert (#7900)

* Revert "Update to dsmr_parser supporting v5 arguments."

This reverts commit 3567de4b90361c262c4d3261325c2b7dbfa5441f.

* Revert "Using dev branch until released upstream."

This reverts commit 53e8de112cb0c47ac5031aacd74a5b18b4a418e2.

* Revert "Give good example."

This reverts commit 4f90fc4be677634fc7e951b4b79aa57c8213b8f8.

* Revert "Allow configuring DSMR5 protocol."

This reverts commit 9fa0e14187e32521e2f51cf2183a9070ee093bd0.

aequitas committed:
Dsmr5 revert (#7900)

* Revert "Update to dsmr_parser supporting v5 arguments."

This reverts commit 3567de4b90361c262c4d3261325c2b7dbfa5441f.

* Revert "Using dev branch until released upstream."

This reverts commit 53e8de112cb0c47ac5031aacd74a5b18b4a418e2.

* Revert "Give good example."

This reverts commit 4f90fc4be677634fc7e951b4b79aa57c8213b8f8.

* Revert "Allow configuring DSMR5 protocol."

This reverts commit 9fa0e14187e32521e2f51cf2183a9070ee093bd0.

mje-nz committed:
Fix typos in Wunderground component (Percipitation -> Precipitation) (#7901)

mje-nz committed:
Fix typos in Wunderground component (Percipitation -> Precipitation) (#7901)

perosb committed:
Add support for the expirationTime parameter. (#7895)

Enabled by default in Chrome 60.
Only accepts the param, doesn't act on the actual expiration date. Chrome will always pass NULL for now.

https://github.com/w3c/push-api/pull/248
https://www.chromestatus.com/feature/4929396687241216
https://bugs.chromium.org/p/chromium/issues/detail?id=718837

nkgilley committed:
Etherscan.io sensor (#7855)

* etherscan sensor

* Update etherscan.py

nkgilley committed:
blockchain.info sensor (#7856)

* blockchain sensor

* Update blockchain.py

* Update blockchain.py

* add validation of btc addresses

cribbstechnologies committed:
Mqtt cover modifications (#7841)

* adding set position ability
removing command_topic being required

* flaking

* flaking test

* updating docs

* requested updates

* Revert "updating docs"

This reverts commit 9cfc5ed7a8f5761de93d8bc7c1cfb81fa8d7ca40.

* forgot to update constructor calls in tests

cribbstechnologies committed:
Mqtt cover modifications (#7841)

* adding set position ability
removing command_topic being required

* flaking

* flaking test

* updating docs

* requested updates

* Revert "updating docs"

This reverts commit 9cfc5ed7a8f5761de93d8bc7c1cfb81fa8d7ca40.

* forgot to update constructor calls in tests

tboyce021 committed:
Add Radarr sensor (#7318)

* Add radarr.py

* Update radarr.py

* Update radarr.py

* Add test_radarr.py

* Update test_radarr.py

* Update test_radarr.py

* Update radarr.py

* Update .coveragerc

* Fix hound.

mattsch committed:
Add service to set nest away/home modes (#7619)

* Add service to set nest away/home modes

* New service `nest.set_mode`
* Update the NestDevice object to export the local structures

* Validation and structure cleanup

mjj4791 committed:
Added buienradar sensor and weather (#7592)

* Added buienradar sensor and weather

* used external library for parsing

* used external library for parsing

* updated buienradar lib to 0.4

* Make sure you import 3rd party libraries inside methods.

* Make sure you import 3rd party libraries inside methods.

* clean up code; optimized

* imports, sensor name and attributes

* updated requirements to match imports

* use asyncio for http get

mbrrg committed:
Add support for Vanderbilt SPC alarm panels and attached sensors (#7663)

* Add support for Vanderbilt SPC alarm panels.

 * Arm/disarm + read state

 * Autodiscover and add motion sensors

* Fix code formatting.

* Use asyncio.async for Python < 3.4.4.

* Fix for moved aiohttp exceptions.

* Add docstrings.

* Fix tests and add docstrings.

balloob committed:
Update .coveragerc

florincosta committed:
Add raspihats switch (#7665)

balloob committed:
Lint

PhracturedBlue committed:
Add 'icon_template' to switch templates (similar to sensor template) (#7862)

* Add 'icon_template' to switch templates (similar to sensor template)

* Add test for template switch 'icon_template'

fabaff committed:
Fix docstring (#7907)

fabaff committed:
Sync crypto-currency platforms (#7906)

fabaff committed:
Move consts to 'const.py' (#7909)

PhracturedBlue committed:
Minor cleanup - Define 'CONF_ICON_TEMPLATE' constant centrally (#7910)

* Add 'icon_template' to switch templates (similar to sensor template)

* Add test for template switch 'icon_template'

* Define 'CONF_ICON_TEMPLATE' constant centrally

* Missed a redundant definition

fabaff committed:
Do not call update() in constructor (#7912)

* Do not call update() in constructor

* Do not call update() in constructor

* Remove unused import

Commit:
Add juicenet platform (#7668)

* Add juicenet platform

* Update missing variable
Add missing blank lines

* Remove unnecessary override

* Update juicenet.py

* Remove whitespace
Add missing docstring

* Remove unused services
Use the hass built in unique_id

* Fix lint issues

* Update python-juicenet library version

* Update python-juicenet library version

* Remove unnecessary code

* Remove unused import

* Remove super call

mezz64 committed:
Bump pyEight version to fix 0hr session errors (#7916)

perosb committed:
update to pywebpush 1.0.4 which allows install on system with openssl-1.1.0 (cryptography dep) (#7915)

fabaff committed:
Do not call update() in constructor (#7917)

fabaff committed:
Fix changes introduced with #7917 (#7930)

fabaff committed:
Do not call update() in constructor (#7931)

mpstephana committed:
Add service_url config option to volvooncall (#7919)

* Add service_url config option to volvooncall

* Import default value from volvooncall lib

jminn committed:
Added 'change' field to statistics sensor (#7820)

* Added 'change' field to statistics sensor

* Updated statistics sensor test

* Updated statistics sensor test complaint

nkgilley committed:
add ripple sensor (#7935)

fabaff committed:
Add Yahoo! weather platform (#7939)

bjw-s committed:
[media_player.sonos] Send media_stop on turn_off (#7940)

danielperna84 committed:
Updated dependency

danielperna84 committed:
Merge pull request #7948 from danielperna84/HomeMatic

HomeMatic: Updated dependency

joopert committed:
update to 006 (#7945)

vrs01 committed:
Update ping.py (#7944)

scarface-4711 committed:
Add option to display all input sources / Add support for favourite channels / Treat Marantz SR5008 as Denon AVR-X device (#7949)

* add option to display all sources / pushed to version 0.4.3 of library

* Add show_all_sources option for auto discovery too

* change code style for hass

* fix lint

Teagan42 committed:
[WIP] Fix opencv (#7864)

* Updates to opencv image processor

* Remove opencv hub

* Requirements

* Remove extra line

* Fix linting errors

* Indentation

* Requirements

* Linting

* Check for import on platform setup

* Remove opencv requirement

* Linting

* fix style

* fix lint

fabaff committed:
Upgrade coinmarketcap to 3.0.1 (#7951)

heinemml committed:
test connection without needing admin rights (#7947)

SHOW DIAGNOSTICS always needs admin privileges on influxdb. For
the purposes of home-assistant this is too much.
Use 'SHOW SERIES' to have a relatively lightweight query which
only needs READ privileges.

bokub committed:
Fix the negative values bug in history_stats (#7934)

alanfischer committed:
Vera colored light support (#7942)

* Added support for color to vera lights

* Updated requirements

fabaff committed:
Fix typos (#7957)

Add an optional extended description…

commento committed:
Osram lightify, removed double set to the lightify bridge in case of brightness changes (#7662)

* osram lightify removed duplicated set in case of brightness changes

* lightify component: anticipate brightness evaluation to handle unconsidered scenario described in the PR request comments (light turned on with color/temperature)

* Correction for travis ci error:
undefined name 'transition'

fabaff committed:
Add configuration check and use default var names (#7963)

trisk committed:
lock.sesame: Update pysesame, add state attributes (#7953)

* Update pysesame requirement to 0.1.0 to support caching

* Set `available` property based on API enabled status

* Add state attributes for device ID and battery level

balloob committed:
Version bump to 0.46.1

Commit:
Fixed metadata issue (#7932)

Commit:
Fixed metadata issue (#7932)

girlpunk committed:
Create metoffice.py (#7965)

Fix met office sensor

girlpunk committed:
Create metoffice.py (#7965)

Fix met office sensor

balloob committed:
Prevent Roku doing I/O in event loop (#7969)

balloob committed:
Prevent Roku doing I/O in event loop (#7969)

balloob committed:
Merge pull request #7968 from home-assistant/release-0-46-1

0.46.1

balloob committed:
Merge branch 'master' into dev

balloob committed:
Fix platforms being able to block startup (#7970)

balloob committed:
New component: Python Script (#7950)

* Add initial version

* Fix requirements

* Prefer logging over printing

* Set executor thread name on >Py36 only

* Add tests

* Lint

* Add restrictedpython to test dependencies

* Create python_script.py

From doc:
```
However, an empty dict ({}) is treated as is. If you want to specify a list that can contain anything, specify it as dict:
>>> schema = Schema({}, extra=ALLOW_EXTRA)  # don't do this
>>> try:
...   schema({'extra': 1})
...   raise AssertionError('MultipleInvalid not raised')
... except MultipleInvalid as e:
...   exc = e
>>> str(exc) == "not a valid value"
True
>>> schema({})
{}
>>> schema = Schema(dict)  # do this instead
>>> schema({})
{}
>>> schema({'extra': 1})
{'extra': 1}

```

Commit:
Nadtcp component (#7955)

* initial commit

* class name and requirements_all.txt

* removed mentions of D7050

* changed default name

* catch oserror in update, travis errors.

* use nad_receiver pip version

* update coveragerc

danielperna84 committed:
HomeMatic optimizations and code cleanup (#7986)

* Cleanup and optimizations

* Cleanup

* Typo -.-

* Linting

molobrakos committed:
Update eliqonline.py (#7977)

Print error

chilicheech committed:
test that all lights turn off when no entity id is given (#7981)

azogue committed:
make `last_name` field optional (#7988)

scarface-4711 committed:
Bugfixing with version 0.4.4 of denonavr (#7995)

fabaff committed:
Upgrade py-cpuinfo to 3.3.0 (#7992)

thomasdelaet committed:
fix issues raised in pull request review

thomasdelaet committed:
Merge branch 'velbus' of github.com:thomasdelaet/home-assistant into velbus

thomasdelaet committed:
specify Exception type

thomasdelaet committed:
add log statement when message is received

thomasdelaet committed:
Update velbus.py

remove exception since potential exception is only thrown asynchronous
